src/gui/editorwidgets/qgsexternalresourceconfigdlg.cpp

```diff
@@ -36,14 +40,21 @@ QgsExternalResourceConfigDlg::QgsExternalResourceConfigDlg( QgsVectorLayer *vl,
 
   mRootPath->setPlaceholderText( QgsSettings().value( QStringLiteral( "/UI/lastExternalResourceWidgetDefaultPath" ), QDir::toNativeSeparators( QDir::cleanPath( defpath ) ) ).toString() );
 
-  // Add connection to button for choosing default path
-  connect( mRootPathButton, SIGNAL( clicked() ), this, SLOT( chooseDefaultPath() ) );
+  QMenu *rootPathMenu = new QMenu();
+  mRootPathExpressionButton = new QAction( QgsApplication::getThemeIcon( QStringLiteral( "/mIconExpression.svg" ) ), tr( "Expression" ), rootPathMenu );
```

Comment:
I was thinking about using the data defined button such as in label
![image](https://cloud.githubusercontent.com/assets/127259/23609552/dba37e58-026e-11e7-89c3-eb742df47bc0.png)

any reason why not to do so?